addons/account/wizard/account_payment_register.py

```diff
@@ -597,23 +585,78 @@ def _create_payments(self):
                         (1, debit_lines[0].id, {'debit': debit_lines[0].debit + delta_balance}),
                         (1, credit_lines[0].id, {'credit': credit_lines[0].credit + delta_balance}),
                     ]})
+        return payments
+
+    def _post_payments(self, to_process, edit_mode=False):
+        """ Post the newly created payments.
 
+        :param to_process:  A list of python dictionary, one for each payment to create, containing:
+                            * create_vals:  The values used for the 'create' method.
+                            * to_reconcile: The journal items to perform the reconciliation.
+                            * batch:        A python dict containing everything you want about the source journal items
+                                            to which a payment will be created (see '_get_batches').
+        :param edit_mode:   Is the wizard in edition mode.
+        """
+        payments = self.env['account.payment']
+        for vals in to_process:
+            payments |= vals['payment']
         payments.action_post()
 
-        domain = [('account_internal_type', 'in', ('receivable', 'payable')), ('reconciled', '=', False)]
-        for payment, lines in zip(payments, to_reconcile):
+    def _reconcile_payments(self, to_process, edit_mode=False):
+        """ Reconcile the payments.
 
-            # When using the payment tokens, the payment could not be posted at this point (e.g. the transaction failed)
-            # and then, we can't perform the reconciliation.
-            if payment.state != 'posted':
-                continue
```

Comment:
@smetl theses lines are left out after the refactoring, and I don't see any filter to not run `_reconcile_payments` on unposted payments?

Reply:
@hoangtrann You are right. This is a bug